runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -228,15 +230,19 @@ public void write(IToken token, Object data) throws OverwriteException {
                     .peek(e, address));
     }
 
+    public @NonNull ILogData read(long address) {
+        return read(address, defaultReadOptions);
+    }
+
     /**
      * Read the given object from an address and streams.
      *
      * @param address An address to read from.
      * @return A result, which be cached.
      */
     public @Nonnull
-    ILogData read(long address) {
-        if (!runtime.getParameters().isCacheDisabled()) {
+    ILogData read(long address, ReadOptions options) {
+        if (!runtime.getParameters().isCacheDisabled() && options.isClientCachable()) {
```

Comment:
For debug, should we consider printing a warning if client is cachable but cache is disabled? it could give some hint of performance impact if we believe the client is configured to cache entries.

Reply:
done.